### PR TITLE
Port allmydata.web to Python 3, part 2

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3611

### CODE REVIEW

**Review by chadwhitacre — CHANGES_REQUESTED** [review]
Switching from `%s` to `%r` changes the result, correct? Are we okay with that?

```
$ (echo 'py t1 t2 %s %r'; echo '-- -- -- -- --'; for py in 27 36; do for t1 in b u; do for t2 in b u; do (echo -n $py $t1 $t2 ''; for s in s r; do echo -n $(.tox/py$py/bin/python -c "print($t1'%$s' % $t2'foo')" 2>/dev/null || echo 💥) ''; done); echo; done; done; done) | column -t
py  t1  t2  %s      %r
--  --  --  --      --
27  b   b   foo     'foo'
27  b   u   foo     u'foo'
27  u   b   foo     'foo'
27  u   u   foo     u'foo'
36  b   b   b'foo'  b"b'foo'"
36  b   u   💥      b"'foo'"
36  u   b   b'foo'  b'foo'
36  u   u   foo     'foo'
$
```

File: src/allmydata/web/logs.py [hunk 1]
```diff
@@ -21,11 +21,6 @@
     Resource,
 )
 
-# Hotfix work-around https://github.com/crossbario/autobahn-python/issues/1151
-from . import _autobahn_1151
-_autobahn_1151.patch()
-del _autobahn_1151
-
```
[the remark this answers is not in the document]
Comment (itamarst) [reply src/allmydata/web/logs.py:28]: I went and checked, the fix was merged into upstream soon after, and Tahoe-LAFS is using a version of Autobahn released after that.

File: tox.ini [hunk 2]
```diff
@@ -71,6 +71,10 @@ commands =
          coverage: coverage xml
          coverage: coverage report
 
+         # We also run tests with -bb to catch bugs like `"%s" % (some_bytes,)`.
+         # Eventually everything should run with this, but so far only parted
+         # some of the modules.
+         python -bb -m twisted.trial --rterrors allmydata.test.web
```
[the remark this answers is not in the document]
Comment (itamarst) [reply tox.ini:77]: Should've been "ported".

**Comment by itamarst** [comment]
All the changes from `%s` to `%r` are logging or error messages, unless I missed one?

File: tox.ini [hunk 3]
```diff
@@ -71,6 +71,10 @@ commands =
          coverage: coverage xml
          coverage: coverage report
 
+         # Also run tests with -bb to catch bugs like `"%s" % (some_bytes,)`.
+         # Eventually everything should run with this, but so far only fixed
+         # some of the Python3-ported modules.
+         python -bb -m twisted.trial --rterrors allmydata.test.web
```
Comment (exarkun) [inline tox.ini:77]: Cool.

I'm thinking about how we can scale this check up further.  Passing with "-bb" is related to but separate from "passes on Python 3" right?  So we can't just split the `PORTED_TEST_MODULES` list into two pieces.  Would it make sense to have a separate `BB_TEST_MODULES` list where test modules that pass w/ -bb get added?  

What might really be nice is if we could run each test just once in the strictest configuration it is known to pass.  That is, create a `BB_TEST_MODULES` list and subtract it from `TEST_SUITE`.  That suddenly creates a lot more complexity though :(

Maybe this is fine as-is for now.  We hope to have someone joining soon who will focus on testing improvements.  One such improvement could be sorting out this mess - or implementing some plan, if you feel like you have an idea of what might make sense to implement.  If so, please write it up in a ticket and link to it from here.

**Review by exarkun — APPROVED** [review]
Thanks.  Porting looks good.  One comment about testing strategy inline.  Address and merge.  Thanks.

File: tox.ini (hunk 3 above)
Comment (itamarst) [reply tox.ini:77]: My experience was that fixing -bb for web wasn't _that_ much work, so might be able to just... enable it for all Python 3 tests withouts too much effort in one PR. So I filed ticket for that and then try that:

https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3618

**Pull request merged**